lib/msal-node/src/client/ManagedIdentityApplication.ts

```diff
@@ -0,0 +1,179 @@
+import { NodeStorage } from "../cache/NodeStorage";
+import { ManagedIdentityResult } from "../response/ManagedIdentityResult";
+import { ManagedIdentityUtils } from "../utils/ManagedIdentityUtils";
+
+/**
+ * Class to initialize a managed identity and identify the service
+ */
+export class ManagedIdentityApplication {
+    private config: ManagedIdentityNodeConfiguration;
+    // needed for unit test
+    public get getConfig(): ManagedIdentityNodeConfiguration {
+        return this.config;
+    }
+
+    private logger: Logger;
+    private cacheManager: CacheManager;
+    private networkClient: INetworkModule;
+    private cryptoProvider: CryptoProvider;
+
+    // authority needs to be faked to re-use existing functionality in msal-common: caching in responseHandler, etc.
+    private fakeAuthority: Authority;
+
+    // the ClientCredentialClient class needs to be faked to call it's getCachedAuthenticationResult method
+    private fakeClientCredentialClient: ClientCredentialClient;
+
+    constructor(configuration?: ManagedIdentityConfiguration) {
+        // undefined config means the managed identity is system-assigned
+        this.config = buildManagedIdentityConfiguration(configuration || {});
+
+        this.logger = new Logger(
+            this.config.system.loggerOptions,
+            name,
+            version
+        );
+
+        const fakeStatusAuthorityOptions: StaticAuthorityOptions = {
+            knownAuthorities: [Constants.DEFAULT_AUTHORITY],
+        };
+        this.cacheManager = new NodeStorage(
+            this.logger,
+            this.config.managedIdentityId.getId,
+            DEFAULT_CRYPTO_IMPLEMENTATION,
+            fakeStatusAuthorityOptions
+        );
+
+        this.networkClient = this.config.system.networkClient;
+
+        this.cryptoProvider = new CryptoProvider();
+
+        const fakeAuthorityOptions: AuthorityOptions = {
+            protocolMode: ProtocolMode.AAD,
+            knownAuthorities: [Constants.DEFAULT_AUTHORITY],
+            cloudDiscoveryMetadata: "",
+            authorityMetadata: "",
+        };
+        this.fakeAuthority = new Authority(
+            Constants.DEFAULT_AUTHORITY,
+            this.networkClient,
+            this.cacheManager,
+            fakeAuthorityOptions,
+            this.logger
+        );
+
+        this.fakeClientCredentialClient = new ClientCredentialClient({
+            authOptions: {
+                clientId: this.config.managedIdentityId.getId,
+                authority: this.fakeAuthority,
+            } as AuthOptions,
+        } as ClientConfiguration);
+    }
+
+    /**
+     * Acquire an access token from the cache or the managed identity
+     * @param managedIdentityRequest
+     * @returns the access token
+     */
+    public async acquireToken(
+        managedIdentityRequestParams: ManagedIdentityRequestParams
+    ): Promise<ManagedIdentityResult> {
```

Comment:
Shouldn't this return `AuthenticationResult` like other APIs?

Reply:
No, a managed identity result has been created which will be returned to the developer.